In [2]:
import numpy as np
sinav_notlari = np.array([80,83,75,90,98,65])
ort_sinav_notlari = np.mean(sinav_notlari)
std_sinav_notlari = np.std(sinav_notlari)
en_dusuk_zskor = (min(sinav_notlari) - ort_sinav_notlari) / (std_sinav_notlari)
en_yuksek_zskor = (max(sinav_notlari) - ort_sinav_notlari) / (std_sinav_notlari)
print(f"Notların ortalaması: {ort_sinav_notlari}")
print(f"Notların standart sapması: {std_sinav_notlari}")
print(f"En düşük not: {min(sinav_notlari)} ve bu nota ait Z-Skor: {en_dusuk_zskor}")
print(f"En yüksek not: {max(sinav_notlari)} ve bu nota ait Z-Skor: {en_yuksek_zskor}")

Notların ortalaması: 81.83333333333333
Notların standart sapması: 10.510576683618375
En düşük not: 65 ve bu nota ait Z-Skor: -1.6015613453036794
En yüksek not: 98 ve bu nota ait Z-Skor: 1.5381331732124455


Elimizdeki notları incelediğimizde en yüksek not 98 ve bu not yaklaşık olarak ortalamadan 1.54 standart sapma yukarıda iken en düşük not (65) ortalamadan 1.60 standart sapma aşağıdadır. Ayrıca yukarıda elle yapmış olduğumuz hesabı SciPy kütüphanesinin Stats modülünden **zscore** fonksiyonu kullanılabilir. Aşağıda bu fonksiyonun kullanımına ait örnek verilmiştir bu fonksiyon varsayılan olarak elimizdeki verinin popülasyon olduğunu kabul eder eğer veri bir örneklem ise serbestlik derecesi ddof = 1 olarak belirlenmelidir. Bizim örneğimizde 6 kişilik grup bir popülasyonu temsil ettiğini varsayarak ddof'u varsayılan ayarıyla bıraktık. Son olarak yukarıdaki her iki z skoru da aykırı değer eşiğini geçmediği için bu veri setinde aykırı değer olmadığı kanaatine varabiliriz.

In [3]:
from scipy.stats import zscore
z_skorlari = zscore(a=sinav_notlari)
print(z_skorlari)

[-0.17442747  0.1109993  -0.65013876  0.77699511  1.53813317 -1.60156135]
